src/main/scala/wandou/avpath/Evaluator.scala

```diff
@@ -445,12 +474,13 @@ object Evaluator {
           }
         }
 
-      case Ctx(map: java.util.Map[String, _] @unchecked, schema, topLevelField, _) =>
+      case Ctx(map: java.util.Map[String, _] @unchecked, name, schema, topLevelField, path, _) =>
         getValueType(schema) foreach { elemType =>
           val entries = map.entrySet.iterator
           while (entries.hasNext) {
             val entry = entries.next
-            val elemCtx = Ctx(entry.getValue, elemType, topLevelField, Some(TargetMap(map, entry.getKey, schema)))
+            // TODO not sure
```

Comment:
We are on a Map, we name each sub element like the parent.
It should not be troublesome, but it's not right.

Reply:
```suggestion
            // We are on a Map, we name each sub element like the parent.  It should not be troublesome, but it's not right.
```